Sources/Extensions/Alamofire/AlamofireManager+Extensions.swift

```diff
@@ -53,71 +52,33 @@ public extension Reactive where Base: Alamofire.SessionManager {
     /// Method that executes request and serializes response into target object
     ///
     /// - Parameter requestParameters: api parameters to pass Alamofire
+    /// - Parameter decoder: json decoder to decode response data
     /// - Parameter mappingQueue: The dispatch queue to use for mapping
     /// - Returns: Observable with HTTP URL Response and target object
-    func responseModel<T: ImmutableMappable>(requestParameters: ApiRequestParameters,
-                                             mappingQueue: DispatchQueue = .global(),
-                                             acceptableStatusCodes: Set<Int> = Base.defaultAcceptableStatusCodes)
+    func responseModel<T: Decodable>(requestParameters: ApiRequestParameters,
+                                     decoder: JSONDecoder,
+                                     mappingQueue: DispatchQueue = .global(),
+                                     acceptableStatusCodes: Set<Int> = Base.defaultAcceptableStatusCodes)
```

Comment:
it's time to add custom SessionManager to avoid code duplication and parameters hell:

```swift
open class SessionManager: Alamofire.SessionManager {

    var mappingQueue: DispatchQueue = .global()
    var decoder: JSONDecoder = JSONDecoder()
    var acceptableStatusCodes = Alamofire.SessionManager.defaultAcceptableStatusCodes

}
```